File: rb/features.rb
# Freepublicapis SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FreepublicapisFeatures
  def self.make_feature(name)
    case name
    when "base"
      FreepublicapisBaseFeature.new
    when "ratelimit"
      FreepublicapisRatelimitFeature.new
    when "retry"
      FreepublicapisRetryFeature.new
    when "test"
      FreepublicapisTestFeature.new
    when "timeout"
      FreepublicapisTimeoutFeature.new
    else
      FreepublicapisBaseFeature.new
    end
  end
end

File: py/features.py
# Freepublicapis SDK feature factory

from feature.base_feature import FreepublicapisBaseFeature
from feature.test_feature import FreepublicapisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreepublicapisBaseFeature(),
        "test": lambda: FreepublicapisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
